### PR TITLE
Add new endpoint for atomically updating multiple dedup results

### PR DESCRIPTION
**Pull request by fabubaker**
rayyan-store server now has an endpoint called `batch-update-dedup` allowing for an atomic batch update of dedup results. If any one transaction in the batch fails, all changes are rolled back.

The corresponding client call has been implemented in the rayyan-store ruby client as `batch_update_dedup_user_action`.

### CODE REVIEW

File: server/src/adapters/postgres.js [hunk 1]
```diff
@@ -1004,5 +1004,61 @@ module.exports = {
...
+    // @param {String} reviewId - The id of the review to update.
+    // @param {Array} updatesList - Array of Dicts of the form
+    //                               {article_id: <article id>,
+    //                               user_action: <new user action>}
+
+    //       WITH update as (SELECT * FROM jsonb_array_elements('${JSON.stringify(updatesList)}'::jsonb))
+
+    const dedupQuery =
+      `
+      WITH update as (SELECT * FROM jsonb_array_elements($1))
```
Comment (hammady) [inline server/src/adapters/postgres.js]: With this query, you don't even need a transaction. The transaction is needed if you are doing separate SQL statements. I like that way that this is done using pure SQL, so it is guaranteed to be atomic, as far as I understand.
However, I like the transaction code, we can abstract it as helper functions to use it elsewhere.

File: server/src/adapters/postgres.js [hunk 2]
```diff
@@ -1004,5 +1004,61 @@ module.exports = {
...
+                })
+                .catch(reject)
+            })
+            .catch(function (_) {
+              abortTransaction(client)
+            })
+        })
+        .catch(function (err) {
+          reject(err)
+        })
```
Comment (hammady) [inline server/src/adapters/postgres.js]: I notice here that `reject` is only called on connection error or query error, but not on `BEGIN`  or `COMMIT`. Conversely, the `abortTransaction` is called in the opposite cases. Is there a reason for that?

File: server/src/adapters/postgres.js (hunk 1 above)
Comment (fabubaker) [reply server/src/adapters/postgres.js]: Just to double-check, [this](https://malisper.me/transactions-in-postgres/) and some manual testing confirms that the statement is indeed atomic and we can remove the transaction code.

File: server/src/adapters/postgres.js (hunk 2 above)
Comment (fabubaker) [reply server/src/adapters/postgres.js]: No reasoning, I misunderstood the exception catching and it should be fixed. But with the new commit removing transactions this is not needed.

**Pull request merged**